### PR TITLE
log: add new log package that uses slog as global logger

### PR DESCRIPTION
**Pull request by jmontesi**
The log package provides top-level funtions for creating log records in four different log levels as well as some other helper functions to set up the logging parameters and create special log handlers.

Two slog handlers are created: the `MultiHandler` that allows distributing the logs to several io.Writer objects and the `CustomHandler` to produce the log output in a specifc format.

The `checksdb.Check` type is also modified to add two new fields: a logger and a log archive where the check's log will be stored. The check's logger has the two handlers described above chained, so that the log output is written to the log file and to the check's log archive.

A log file has been created with a default name _cnf-test-log_.

Only a couple of log lines have been produces using this new logger as an example. The bulk of the work for converting the old log lines so that they use the new package will be done in a separate change.

### CODE REVIEW

File: pkg/certsuite/certsuite.go [hunk 1]
```diff
@@ -14,6 +21,16 @@ import (
...
+func LoadChecksDB() {
+	accesscontrol.LoadChecks()
+	certification.LoadChecks()
+	lifecycle.LoadChecks()
+	manageability.LoadChecks()
+	networking.LoadChecks()
+	observability.LoadChecks()
+	performance.LoadChecks()
+}
+
```
Comment (sebrandon1) [inline pkg/certsuite/certsuite.go:33]: @jmontesi Could the `LoadChecks` changes be in a separate PR?  I love the idea of getting rid of those Init() funcs and it doesn't look like its tied to the logging changes.

Comment (jmontesi) [reply pkg/certsuite/certsuite.go:33]: Sure, here's the [PR](https://github.com/test-network-function/cnf-certification-test/pull/1664).

File: internal/log/custom_handler.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+func (h *CustomHandler) Enabled(_ context.Context, level slog.Level) bool {
+	return level >= h.opts.Level.Level()
+}
+
+// The Handle method will write a log line with the following format:
+// LOG_LEVEL [TIME] [SOURCE_FILE] [CUSTOM_ATTRS] MSG
+//
+//nolint:gocritic
```
Comment (greyerof) [inline internal/log/custom_handler.go:46]: Please add a comment on why this gocritic is needed.

File: internal/log/custom_handler.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+func (h *CustomHandler) Handle(_ context.Context, r slog.Record) error {
+	buf := make([]byte, 0, 1024) //nolint:gomnd
```
Comment (greyerof) [inline internal/log/custom_handler.go:48]: Taking into account that we shouldn't be worried about performance/speed, both `var buf []byte` or `buf := []byte{}` options should also work here and they wouldn't need the `nolint`.

File: main.go [hunk 4]
```diff
@@ -50,6 +45,8 @@ const (
 	TNFReportKey                  = "cnf-certification-test"
 	extraInfoKey                  = "testsExtraInfo"
 	noLabelsExpr                  = "none"
+	logFileName                   = "cnf-test-log"
```
Comment (greyerof) [inline main.go:48]: I know it's just a sample file name, but I believe it should have an extension. My proposal for a default name is `cnf-certsuite.log`.

**Review by greyerof — APPROVED** [review]
Looks good. I just left some suggestions.

File: internal/log/custom_handler.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	buf = append(buf, "\n"...)
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	_, err := h.out.Write(buf)
+	return err
+}
+
+// Not implemented. WithGroup just returns the same handler.
+func (h *CustomHandler) WithGroup(_ string) slog.Handler {
+	return h
```
[the remark this answers is not in the document]
Comment (jmontesi) [reply internal/log/custom_handler.go:76]: I don't think it's a good idea to log inside a Handler. I will return nil instead.

File: internal/log/custom_handler.go (hunk 2 above)
Comment (jmontesi) [reply internal/log/custom_handler.go:46]: Comment added.

File: internal/log/custom_handler.go (hunk 3 above)
Comment (jmontesi) [reply internal/log/custom_handler.go:48]: Done.

File: main.go (hunk 4 above)
Comment (jmontesi) [reply main.go:48]: Done.
